PyDSS/dssInstance.py

```diff
@@ -558,6 +564,47 @@ def _UpdatePlots(self):
             self._pyPlotObjects[Plot].UpdatePlot()
         return
 
+    def _GetActiveScenario(self):
+        active_scenario = self._Options["Project"]["Active Scenario"]
+        for scenario in self._Options["Project"]["Scenarios"]:
+            if scenario["name"] == active_scenario:
+                return scenario
+        raise InvalidConfiguration(f"Active Scenario {active_scenario} is not present")
+
+    def _SetSimulationTimeBasedOnMode(self, scenario):
+        """Adjusts the time parameters based on the mode."""
+        p_settings = self._Options["Project"]
+        config = scenario["Auto snapshot start time config"]
+        mode = config["mode"]
+        assert mode != SimulationTimeMode.NONE, mode
+
+        if mode == SimulationTimeMode.MAX_PV_LOAD_RATIO:
+            if p_settings["Simulation Type"] != "QSTS":
+                raise InvalidConfiguration(f"{mode} is only supported with QSTS simulations")
+
+            # These settings have to be temporarily overridden because of the underlying
+            # implementation to create a load shape dataframes..
+            orig_start = p_settings["Start time"]
+            orig_duration = p_settings["Simulation duration (min)"]
+            if orig_duration != p_settings["Step resolution (sec)"] / 60:
+                raise InvalidConfiguration("Simulation duration must be the same as resolution")
+            try:
+                p_settings["Start time"] = config["start_time"]
+                p_settings["Simulation duration (min)"] = config["search_duration_min"]
+                new_start = get_snapshot_timepoint(self._Options).strftime(DATE_FORMAT)
+                p_settings["Start time"] = new_start
+                self._Logger.info("Changed simulation start time from %s to %s",
+                    orig_start,
+                    new_start,
+                )
+            except Exception:
+                p_settings["Start time"] = orig_start
+                raise
```

Comment:
nit: as it raises, maybe `p_settings` update could be ignored?

Reply:
But someone else could catch the exception at a higher level. My thought was to ensure that the contents of the global settings object is correct in that case.